### ISSUE TITLE
apply encryption retroactively with filter-branch

### ISSUE THREAD

**Issue by Aeon** [AGWA/git-crypt#7]
Would be nice to have an option in git-crypt to allow one to encrypt the protected files retroactively starting from beginning of time. This would rewrite all history, so would have to be done with care, but would be quite useful when doing an initial public release of a private repository. 

Is this possible to do manually?

**Comment by nanodust** [comment]
new to git-crypt. seems to work well... assuming it's a new repo :) 
this feature would still be quite useful.

**Comment by sean9999** [comment]
love the idea and agree with author that it should be a separate script. i'll whip one up in bash

**Comment by sebastianst** [comment]
so anyone has a `git filter-branch` recipe on how to do this gracefully? I'd love to see this added as a subcommand to git-crypt.

**Comment by ioquatix** [comment]
Good idea, took me like an hour to figure out how to do it and it was a nightmare to get it working.. rebasing etc

**Comment by johansja** [comment]
Any step to step guide on how to do this?

**Comment by axelcostaspena** [comment]
Although the procedure seems more or less easy, using filter-branch for adding the gitattributes from the beggining of the project and filter-branch all descendant commits, it seems that when the repo is big it always re-decrypt the files at several intermediate commits :(

**Comment by n8layman** [comment]
Has anyone figured out a git-filter-repo approach to this?
